Java/syntax_test_java.java

```diff
@@ -201,61 +222,97 @@ public void test2(Type) abc
-
-new Foo<? extends Bar>();
-//      ^ keyword.operator.wildcard.java
-//        ^^^^^^^ keyword.declaration.extends.java
-
-new Foo<? super Bar>();
-//      ^ keyword.operator.wildcard.java
-//        ^^^^^ keyword.declaration.extends.java
+public class Lambdas {
+  volatile int foo;
+//^^^^^^^^ storage.modifier.java
+
+  void anonymousFunctions() {
+     foo();
+//   ^^^ variable.function.java
+
+// Capital names are usually used for classes
+     Foo();
+//   ^^^ variable.function.java
+
+     foo ();
+//   ^^^ variable.function.java
+     this.<A>foo();
+//           ^^^ variable.function.java
+//        ^^^ meta.generic.java
+//        ^ punctuation.definition.generic.begin.java
+//         ^ support.class.java
+     this.<B> foo();
+//            ^^^ variable.function.java
+//        ^^^ meta.generic.java
+//        ^ punctuation.definition.generic.begin.java
+//         ^ support.class.java
+
+     Function<String, Integer> func = a -> 42;
+//                                  ^^^^^^^^^ meta.assignment.rhs.java
+//                                    ^ variable.parameter.java
+//                                      ^^ storage.type.lambda.java
+//                                         ^^ constant.numeric
+//                                           ^ punctuation.terminator.java
+     foo(a -> 42);
+//   ^^^^^^^^^^^^ meta.function-call.java
+//   ^^^ variable.function.java
+//      ^ punctuation.section.parens.begin.java
+//       ^ variable.parameter.java
+//         ^^ storage.type.lambda.java
+//            ^^ constant.numeric
+//              ^ punctuation.section.parens.end.java
+//               ^ punctuation.terminator.java
 
-new Foo<int>();
-//      ^^^ invalid.illegal.primitive-instantiation.java
+     a -> { return 42; };
+//        ^^^^^^^^^^^^^^ meta.lambda.body.java
+
+     (a, b) -> 42;
+//    ^ variable.parameter.java
+//       ^ variable.parameter.java
+//          ^^ storage.type.lambda.java
+//             ^^ constant.numeric
+
+     (int a, Foo<Integer>[] b) -> 42;
+//    ^^^ storage.type.primitive
+//        ^ variable.parameter.java
+//           ^^^ support.class.java
+//              ^ punctuation.definition.generic.begin.java
+//               ^^^^^^^ support.class.java
+//                      ^ punctuation.definition.generic.end.java
+//                          ^ variable.parameter.java
+//                             ^^ storage.type.lambda.java
+//                                ^^ constant.numeric
+
+  new Foo<Abc>();
+//       ^^^^^ meta.generic.java
+//        ^^^ support.class.java
+//       ^ punctuation.definition.generic.begin.java
+//           ^ punctuation.definition.generic.end.java
+
+  new Foo<?>();
+//        ^ keyword.operator.wildcard.java
+
+  new Foo<? extends Bar, String>();
+//        ^ keyword.operator.wildcard.java
+//                  ^^^ support.class.java
+//          ^^^^^^^ keyword.declaration.extends.java
+//                     ^ punctuation.section.generic.separator.java
```

Comment:
I think this should probably be `punctuation.separator` or some specialization of that.